File: laondry-order-service/internal/entity/service.go
package entity

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Service struct {
	ID                 uuid.UUID      `gorm:"type:uuid;primary_key" json:"id"`
	CategoryID         uuid.UUID      `gorm:"type:uuid;not null;index" json:"category_id"`
	Code               string         `gorm:"type:varchar(50);not null;unique" json:"code"`
	Name               string         `gorm:"type:varchar(150);not null" json:"name"`
	Description        *string        `gorm:"type:text" json:"description"`
	PricingModel       string         `gorm:"type:varchar(20);not null;index" json:"pricing_model"`
	BasePrice          float64        `gorm:"type:decimal(12,2);default:0" json:"base_price"`
	MinQty             float64        `gorm:"type:decimal(8,2);default:0" json:"min_qty"`
	EstDurationHours   int            `gorm:"default:24" json:"est_duration_hours"`
	IsExpressAvailable bool           `gorm:"default:false" json:"is_express_available"`
	IsActive           bool           `gorm:"default:true;index" json:"is_active"`
	IconPath           *string        `gorm:"type:varchar(255)" json:"icon_path"`
	CreatedBy          *uuid.UUID     `gorm:"type:uuid" json:"created_by"`
	UpdatedBy          *uuid.UUID     `gorm:"type:uuid" json:"updated_by"`
	CreatedAt          time.Time      `gorm:"type:timestamptz;not null" json:"created_at"`
	UpdatedAt          time.Time      `gorm:"type:timestamptz;not null" json:"updated_at"`
	DeletedAt          gorm.DeletedAt `gorm:"type:timestamptz;index" json:"deleted_at,omitempty"`

	Category      *ServiceCategory `gorm:"foreignKey:CategoryID;references:ID" json:"category,omitempty"`
	ServiceAddons []ServiceAddon   `gorm:"foreignKey:ServiceID;constraint:OnDelete:CASCADE" json:"service_addons,omitempty"`
	Prices        []ServicePrice   `gorm:"foreignKey:ServiceID;constraint:OnDelete:CASCADE" json:"prices,omitempty"`
}

func (Service) TableName() string {
	return "services"
}

func (s *Service) BeforeCreate(tx *gorm.DB) error {
	if s.ID == uuid.Nil {
		s.ID = uuid.New()
	}
	return nil
}

File: laondry-system-management-core-api/app/Models/OrderService/Service.php
<?php

namespace App\Models\OrderService;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use App\Models\Auth\User;
use App\Helpers\ImageStorage;

class Service extends Model
{
    use HasFactory, HasUuids, SoftDeletes;

    protected $fillable = [
        'code',
        'category_id',
        'name',
        'description',
        'pricing_model',
        'base_price',
        'min_qty',
        'est_duration_hours',
        'is_express_available',
        'is_active',
        'icon_path',
        'workflow_template_id',
        'created_by',
        'updated_by'
    ];
    protected $casts = [
        'base_price' => 'decimal:2',
        'min_qty' => 'decimal:2',
        'is_express_available' => 'bool',
        'is_active' => 'bool',
    ];

    protected $appends = ['icon_url'];

    public function getIconUrlAttribute(): ?string
    {
        return ImageStorage::publicUrl($this->icon_path);
    }

    public function category()
    {
        return $this->belongsTo(ServiceCategory::class, 'category_id');
    }
    public function creator()
    {
        return $this->belongsTo(User::class, 'created_by');
    }
    public function updater()
    {
        return $this->belongsTo(User::class, 'updated_by');
    }

    // Pivot M2M
    public function addons()
    {
        return $this->belongsToMany(Addon::class, 'service_addons')
            ->withPivot(['is_required', 'created_at', 'updated_at'])
            ->withTimestamps();
    }
    public function serviceAddons()
    {
        return $this->hasMany(ServiceAddon::class);
    }

    public function prices()
    {
        return $this->hasMany(ServicePrice::class);
    }
    public function orderItems()
    {
        return $this->hasMany(OrderItem::class);
    }

    public function workflowTemplate()
    {
        return $this->belongsTo(StatusWorkflowTemplate::class, 'workflow_template_id');
    }

    public function scopeActive($q)
    {
        return $q->where('is_active', true);
    }
}
